Necrosol (and Romerol) resurrect plants from the dead

Mehnix opened the pull request:
Adds the ability for Necrosol (and romerol) to resurrect plants from the dead.

## About the PR
Currently, Necrosol does exactly the same thing cryoxodone does when you put it in a plant, that being de-age it. Which is pretty dissapointing. This PR makes it so adding 4u of Necrosol (or any amount of romerol) will resurrect a plant.

Why also Romerol? Well it's also a chemical that revives people, and much like with people it will horribly mutate the plant in the reviving process.

## Why / Balance
Most botany chemicals can be made pretty early into the shift, this one gives a reward for those thousands of units of omnizine they send to chemistry.

Not particularly overpowered because generally plants only die in botany due to negligence or lack of knowledge/resources and getting seeds for new plants is easy.

Necrosol will also de-age the plant similarly to cryoxadone although weaker, if the plant was young it may de-age right back into a seed.

## Technical details
Adds a new plant-related entityeffect "PlantResurrect", applies it to necrosol and romerol.

Also somewhat related I noticed that presumably since the entity effect moveabout botany chems no longer display how much they need to perform an effect on a plant. IE: In addition to this effect, Phalanxamine also doesn't say it needs 4u to remove unviability.

## Media
https://github.com/user-attachments/assets/5c5c24f8-0c12-4a22-888a-55f77d1aa8a2

## Requirements
- [X] I have read and am following the [Pull Request and Changelog Guidelines](https://docs.spacestation14.com/en/general-development/codebase-info/pull-request-guidelines.html).
- [X] I have added media to this PR or it does not require an in-game showcase.

## Breaking changes
Slightly changes the existing Necrosol and Romerol yamls.

**Changelog**

- add: 4u of Necrosol or any amount of Romerol will now resurrect a plant from the dead at the cost of their seeds.

Mehnix committed:
Rise from your grave!

ScarKy0 commented:
> Phalanxamine is to my knowledge the only chemical requiring a specific amount of it to have an effect. The rest are chance based or constant.

Ah I see. I still think 5u will be better though since necrosol is rare enough as it is, and reviving plants seems very niche.

Mehnix committed:
ARISE!

Necrosol now takes 4u rather than 9u

Princess-Cheeseballs reviewed (changes requested):
Minor grammar thing. 

I'm not a big fan of necrosol personally so making it a pure botany chem used to revive dead plants would be interesting and means I wouldn't have to remove it later down the line.

The only thing I'd say is that it should come with some kind of downside, maybe it decreases the potency of the plant or the mutation multiplier or both?

I'd also want to check with the other botany maintainer as to if they think this would mess up refactors but considering it's contained within entity effects purely, I don't think it will.

Mehnix committed:
Necrosol "debuff" and grammar fix

Necrosol now still has the cryoxadone effect in addition to its healing and resurrection powers. Limiting its usage amounts.

It is both a debuff in the sense you can't spam it as it'll just undo plant growth, but can also be used tactically as the de-aging effect is less potent than that of cryoxadone, allowing you to get more healing out of it.

Mehnix commented:
> The only thing I'd say is that it should come with some kind of downside, maybe it decreases the potency of the plant or the mutation multiplier or both?

I've decided on a different path for debuffing it a bit in simply adding back cryoxadone's de-aging effect, but have it be weaker, activating 50% of the time per unit. This prevents you spamming the reagent as it will de-age the plant but also adds more interesting usage as cryoxadone's de-aging is so potent any more than 2-3 units resets the plant back to a seed. Here adding 5u will send it back to a child more often that not, which still delays the plant's use in harvesting. It's also thematic in sticking with the cryox->deaging thing.

Decrease in potency and mutation modifier would simply mean any botanist would just mix the chemical with robust harvest/left4zed and call it a day to enjoy infinite healing and resurrection. The only chemical that ages a plant is diethylamine, which does so slowly compared to cryox's/necrosol's de-aging.

Princess-Cheeseballs commented on Resources/Prototypes/Reagents/medicine.yml:
```diff
@@ -1308,7 +1308,10 @@
+  - !type:PlantCryoxadone
+    probability: 0.5
+  - !type:PlantResurrect
+    minScale: 4
```
Very minor thing, I'd keep the probability at 100% to keep it consistent, and probably have it also render the plant seedless. It's likely that if you mutate a plant a bunch it won't be viable anyways but this at least ensures it.

Princess-Cheeseballs replied:
In addition remove the healing and toxin removal. You don't need much of this stuff so it should be getting paired with other chems like dylo anyways.

Princess-Cheeseballs reviewed (changes requested):
one thing.

Mehnix committed:
Power Word: Kill (seeds)

resurrection comes at the cost of sterility

Princess-Cheeseballs reviewed (changes requested):
Don't hardcode seed removal into the effect, just add another effect in YAML.

Mehnix committed:
Choose your fate, mortal

seedlessness on resurrection is no longer hardcoded, instead choosable in yaml with reviveSeedless true/false.

Defaults true.

Mehnix commented:
> Don't hardcode seed removal into the effect, just add another effect in YAML.

Fair enough, adding the existing DestroySeedsEffect wouldn't be fit for purpose however as seeds should only be removed specifically when a plant is revived not any time necrosol is used and creating a new effect just for removing seeds for dead plants is clunky so i've just made it toggleable kind of like how Robust Harvest handles its seed removing effect. 

Appreciate the reviews.

Princess-Cheeseballs reviewed (changes requested):
I guess this is fine for now. One last thing.
